src/main/java/com/avispl/symphony/dal/device/axis/m3064/AxisCommunicator.java

```diff
@@ -462,10 +720,74 @@ private void contributeNoneValueForDeviceStatistics(Map<String, String> stats) {
 		stats.put(AxisMonitoringMetric.DEVICE_INFO.getName() + AxisConstant.WEB_URL, AxisConstant.NONE);
 	}
 
+	/**
+	 * Retrieve the protocol from the device, try with http first, if any failed, continue try with https
+	 *
+	 * set the axisProtocol information if retrieving success else is none
+	 */
+	private void retrieveProtocol() {
+		try {
+			axisProtocol = getAxisProtocol();
+		} catch (Exception e) {
+			axisProtocol = AxisConstant.HTTPS;
+			try {
+				axisProtocol = getAxisProtocol();
+			} catch (Exception ex) {
+				throw new ResourceNotReachableException(AxisConstant.ERR_PROTOCOL);
+			}
+		}
+	}
+
+	/**
+	 * Retrieve the protocol configuration of the device (HTTP, HTTPs or both)
+	 *
+	 * @return the protocol information if retrieving success else is none
+	 */
+	public String getAxisProtocol() {
+		try {
+			SystemParameterDefinitions response = doGet(buildDeviceFullPath(AxisURL.URL_PARAM_CGI) + AxisConstant.QUESTION_MARK + AxisPayloadBody.ACTION + AxisPayloadBody.GET_PROTOCOL,
+					SystemParameterDefinitions.class);
+			SystemChildGroupList[] systemChildGroup1 = response.getGroup().getGroupChildren().getSystemChildGroupLists();
```

Comment:
maybe the variable name here could be a bit more clear (just systemChildGroup?), was it autogenerated, or intentional to have '1' at the end? :)